### ISSUE TITLE
Adding first-class support for `.tofu` files.

### ISSUE THREAD

**Issue by skyzyx** [terraform-docs/terraform-docs#811]
### What problem are you facing?

OpenTofu is adding support for `.tofu` files in addition to `.tf` files. Terraform only supports `.tf` files, and OpenTofu will support both. This will allow users to have tool-specific configurations in places where it matters (e.g., provider functions landing in Terraform 1.8 and OpenTofu 1.7, and wanting to gate-keep the CLI tool by version).

If two files have the same name with each file extension (e.g., `main.tf` and `main.tofu`), OpenTofu will execute `main.tofu` and NOT `main.tf`, while Terraform will execute `main.tf`.

It's still all HCL syntax in these files.

### How could terraform-docs help solve your problem?

I don't believe that generating two sets of documentation is the right approach — one for Terraform, and one for OpenTofu. Instead, I believe that in cases where `.tofu` files exist, the OpenTofu configuration should be provided in tandem with the Terraform configuration.

For example, in the JSON output, `requirements` is an array of objects, with each object defining `name` and `version`. Perhaps this could be modified to support `"name": "opentofu"` with its version, and for any providers/modules, there is a new `tool` key which contains an array of tool identifiers (e.g., `terraform` and `opentofu`).

This would allow flexibility in the future in case (God forbid) another fork happens in the community. This also _shouldn't_ (AFAIK) require any backwards-incompatible changes.

This would also mean that for [content](https://terraform-docs.io/user-guide/configuration/content/), it would be useful to define `.IsTerraform` and `.IsOpenTofu` variables, so that users can adjust the examples or other custom documentation to more specifically target each tool.

Ways for readers to switch between the two flavors of tooling _could_ be either a tabbed interface (and a nice touch would be like many API documentation websites where if you select a preferred language in a tabbed interface to view those examples, that all other examples switch to that same language too) — OR something simpler, but less "nice" like a `<select>` HTML field where you can select which version to view.

**Comment by pascal-hofmann** [comment]
I just merged #833 which adds support for .tofu files. I think this is a good first step. Let's see how this works out, and what we need to improve in the wild with projects that used a mix of both file endings.

**Comment by khos2ow** [comment]
@muffl0n , you are not wrong on this. I've spent a bit of time today and as of now `v0.20.0` only partially supports OpenTofu. I've updated [release notes](https://github.com/terraform-docs/terraform-docs/releases/tag/v0.20.0) accordingly.

As of now, we are only able to read header and footer from `.tofu` files. I'm working on it.

As I've looked around, nothing else needs to be done within `terraform-docs/terraform-docs` repo though, `terraform-docs/terraform-config-inspect` needs updating.

I am going to reopen this issue for further tracking and visibility.

**Comment by jamesgeddes** [comment]
Apologies for the omission!

**Comment by mcwarman** [comment]
Here's a reproducer:

```hcl
# main.tofu
locals {
  regions = toset(["us-east-1", "eu-west-1"])
}

provider "aws" {
  for_each = local.regions
  alias    = "main"
  region   = each.key
}

data "aws_caller_identity" "current" {
  for_each = local.regions
  provider = aws.main[each.key]
}

output "aws_caller_identity_arn" {
  value = values(data.aws_caller_identity.current)[*].arn
}
```

**Comment by mcwarman** [comment]
Raised #895
